app/controllers/orders/conversation.js

```diff
@@ -68,8 +69,9 @@ export default detail.extend({
     var message = this.store.createRecord("message", values);
     message
       .save()
-      .then(() => {
+      .then(data => {
         this.set("body", "");
+        this.get("messages").pushObject(data._internalModel);
```

Comment:
Why there is need to get `_internalModel`? I find fetching data from Ember data though `_internalModel` very fishy 😄 .

Reply:
agreed

Reply:
Since pushing `data` directly gives error:
```js
Uncaught TypeError: internalModel.getRecord is not a function
    at Class.objectAtContent (-private.js:9360)
    at Class.objectAt (array_proxy.js:233)
    at objectAt (array.js:42)
    at addObserverForContentKey (array.js:399)
    at EachProxy.arrayDidChange (array.js:353)
    at arrayContentDidChange (array.js:118)
    at Class._emberMetal.Mixin.create._Mixin$create.arrayContentDidChange (array.js:286)
    at Class.arrangedContentArrayDidChange (array_proxy.js:345)
    at Object.applyStr (ember-utils.js:529)
    at sendEvent (ember-metal.js:425)
```
Hence using `(data._internalModel);` which suffices our purpose here.